protocols/gossipsub/src/config.rs

```diff
@@ -101,22 +100,22 @@ pub struct GossipsubConfig {
     published_message_ids_cache_time: Duration,
 }
 
-impl GossipsubConfig {
+impl Config {
     // All the getters
 
     /// The protocol id to negotiate this protocol. By default, the resulting protocol id has the form
-    /// `/<prefix>/<supported-versions>`, but can optionally be changed to a literal form by providing some GossipsubVersion as custom_id_version.
+    /// `/<prefix>/<supported-versions>`, but can optionally be changed to a literal form by providing some Version as custom_id_version.
     /// As gossipsub supports version 1.0 and 1.1, there are two suffixes supported for the resulting protocol id.
     ///
-    /// Calling `GossipsubConfigBuilder::protocol_id_prefix` will set a new prefix and retain the prefix logic.
-    /// Calling `GossipsubConfigBuilder::protocol_id` will set a custom `protocol_id` and disable the prefix logic.
+    /// Calling `Builder::protocol_id_prefix` will set a new prefix and retain the prefix logic.
+    /// Calling `Builder::protocol_id` will set a custom `protocol_id` and disable the prefix logic.
```

Comment:
```suggestion
    /// Calling [`Builder::protocol_id_prefix`] will set a new prefix and retain the prefix logic.
    /// Calling [`Builder::protocol_id`] will set a custom `protocol_id` and disable the prefix logic.
```

Should be a proper type reference.

Reply:
:heavy_check_mark: